tests/backup/backup_kubevirt_test.go

```diff
@@ -1335,3 +1335,217 @@ var _ = Describe("{KubevirtVMBackupRestoreWithNodeSelector}", func() {
+			log.InfoD("SSH into the kubevirt VM to check VM health after freeze rule")
+			ctx, err := backup.GetAdminCtxFromSecret()
+			log.FailOnError(err, "Fetching px-central-admin ctx")
+			for _, vm := range allVMs {
+				log.Infof("Running command for VM [%s]", vm.Name)
+				output, err := RunCmdInVM(vm, "uname -a", ctx)
+				log.InfoD("Output of command in step - [%s]", output)
+				log.FailOnError(err, "Failed to run command in VM")
+			}
+		})
+
+		Step("Taking backup of kubevirt VM without freeze rule and with unfreeze rule", func() {
+			log.InfoD("Taking backup of kubevirt VM without freeze rule and with unfreeze rule")
+			ctx, err := backup.GetAdminCtxFromSecret()
+			log.FailOnError(err, "Fetching px-central-admin ctx")
+			backupName = fmt.Sprintf("%s-%s", "backup-unfreeze-without-freeze", RandomString(6))
+			backupNames = append(backupNames, backupName)
+			var vmNames []string
+			for _, v := range allVMs {
+				vmNames = append(vmNames, v.Name)
+			}
+			log.Infof("VMs to be backed up - %v", vmNames)
+			unfreezeRuleUid, err = Inst().Backup.GetRuleUid(BackupOrgID, ctx, unfreezeRuleName)
+			log.FailOnError(err, "Unable to fetch unfreeze rule uid - %s", unfreezeRuleName)
+			err = CreateVMBackupWithValidation(ctx, backupName, allVMs, SourceClusterName, backupLocationName, backupLocationUID, scheduledAppContexts,
+				labelSelectors, BackupOrgID, sourceClusterUid, "", "", unfreezeRuleName, unfreezeRuleUid, true)
+			dash.VerifyFatal(err, nil, fmt.Sprintf("Verifying creation and validation of VM backup [%s]", backupName))
+		})
+
+		Step("SSH into the kubevirt VM to check VM health after unfreeze rule", func() {
+			log.InfoD("SSH into the kubevirt VM to check VM health after unfreeze rule")
+			ctx, err := backup.GetAdminCtxFromSecret()
+			log.FailOnError(err, "Fetching px-central-admin ctx")
+			for _, vm := range allVMs {
+				log.Infof("Running command for VM [%s]", vm.Name)
+				output, err := RunCmdInVM(vm, "uname -a", ctx)
+				log.InfoD("Output of command in step - [%s]", output)
+				log.FailOnError(err, "Failed to run command in VM")
+			}
+		})
+	})
+
+	JustAfterEach(func() {
+		defer EndPxBackupTorpedoTest(scheduledAppContexts)
+
+		ctx, err := backup.GetAdminCtxFromSecret()
+		log.FailOnError(err, "Fetching px-central-admin ctx")
+		opts := make(map[string]bool)
+		opts[SkipClusterScopedObjects] = true
+
+		RuleEnumerateReq := &api.RuleEnumerateRequest{
+			OrgId: BackupOrgID,
+		}
+		ruleList, err := Inst().Backup.EnumerateRule(ctx, RuleEnumerateReq)
+		for _, r := range ruleList.GetRules() {
+			log.Infof("Deleting rule [%s]", r.Name)
+			_, err := Inst().Backup.DeleteRule(ctx, &api.RuleDeleteRequest{
+				OrgId: BackupOrgID,
+				Name:  r.Name,
+				Uid:   r.Uid,
+			})
+			dash.VerifySafely(err, nil, fmt.Sprintf("Deleting rule [%s]", r.Name))
+		}
+
+		log.Info("Destroying scheduled apps on source cluster")
+		//err = DestroyAppsWithData(scheduledAppContexts, opts, controlChannel, errorGroup)
```

Comment:
Do you really need this code?

Reply:
Yes, currently we do not have the VM data validation part ready. Once we have that I have to replace `ValidateApplications` with `DestroyAppsWithData`

Reply:
Yes